### PR TITLE
✨ Add fiat purchase nft functions

### PR DESCRIPTION
**Pull request by williamchong**
Current design is to have a LIKE granter wallet for all fiat purchase, the wallet would have existing authz spend grant to api wallet. Every time we have a fiat tx received we would run the original flow for purchasing nft using granter wallet.
We would use auth and capture method for payment, payment is captured only when the tx success

### CODE REVIEW

File: src/routes/likernft/fiat/stripe.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+              product_data: {
+                name,
+                description,
+                images: [image],
+                metadata: {
+                  iscnPrefix,
+                  classId,
+                },
+              },
+              unit_amount: new BigNumber(fiatPriceString).shiftedBy(2).toNumber(),
```
Comment (williamchong) [inline src/routes/likernft/fiat/stripe.js]: `toNumber()` round to 0 dp

File: src/util/api/likernft/fiat/index.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+const CURRENCY = 'usd';
+
+let fiatGranterWallet;
+
+async function getLIKEPrice() {
+  const cachedPrice = priceCache.get(CURRENCY);
+  if (cachedPrice) {
+    return cachedPrice;
+  }
+  const price = await Promise.all([
```
Comment (williamchong) [inline src/util/api/likernft/fiat/index.js]: Why array?

File: src/util/api/likernft/fiat/index.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    axios.get(COINGECKO_PRICE_URL)
+      .then((r) => {
+        const p = r.data.market_data.current_price[CURRENCY];
+        priceCache.set(CURRENCY, p);
```
Comment (williamchong) [inline src/util/api/likernft/fiat/index.js:34]: Should compare with min in cache?

File: src/util/api/likernft/fiat/index.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        return p;
+      })
+      .catch(() => LIKER_NFT_FIAT_MIN_RATIO),
```
Comment (williamchong) [inline src/util/api/likernft/fiat/index.js]: Use last price?

File: src/routes/likernft/fiat/stripe.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+      };
+      const { image, name, description } = metadata;
+      const gasFee = getGasPrice();
+      const totalPrice = price + gasFee;
+      const fiatPriceString = await getFiatPriceStringForLIKE(totalPrice);
+      const paymentId = uuidv4();
+      const session = await stripe.checkout.sessions.create({
+        mode: 'payment',
+        success_url: `https://${LIKER_LAND_HOSTNAME}/nft/fiat/stripe?class_id=${classId}&payment_id=${paymentId}`,
+        cancel_url: `https://${LIKER_LAND_HOSTNAME}/nft/${classId}`,
```
Comment (nwingt) [inline src/routes/likernft/fiat/stripe.js]:
```suggestion
        cancel_url: `https://${LIKER_LAND_HOSTNAME}/nft/class/${classId}`,
```

**Pull request merged**